rewrite-java/src/main/java/org/openrewrite/java/controlflow/ControlFlow.java

```diff
@@ -478,24 +487,11 @@ public J.WhileLoop visitWhileLoop(J.WhileLoop whileLoop, P p) {
             Set<ControlFlowNode.ConditionNode> conditionNodes = allAsConditionNodesMissingTruthFirst(conditionAnalysis.current);
             // Then the body is visited
             // Only transfer exits
-            if (whileLoop.getBody() instanceof J.Empty) {
-                if (conditionNodes.size() == 1) {
-                    ControlFlowNode.ConditionNode conditionNode = conditionNodes.iterator().next();
-                    conditionNode.addSuccessor(conditionNode);
-                    current = Collections.singleton(conditionNode);
-                } else {
-                    ControlFlowNode.ConditionNode missingOne = getControlFlowNodeMissingOneSuccessor(conditionNodes);
-                    ControlFlowNode.ConditionNode missingBoth = getControlFlowNodeMissingBothSuccessors(conditionNodes);
-                    missingBoth.addSuccessor(missingOne);
-                    current = conditionNodes;
-                }
-            } else {
-                ControlFlowAnalysis<P> bodyAnalysis = visitRecursiveTransferringExit(conditionNodes, whileLoop.getBody(), p);
-                // Add the 'loop' in
-                bodyAnalysis.current.forEach(controlFlowNode -> controlFlowNode.addSuccessor(entryBlock.getSuccessor()));
-                bodyAnalysis.continueFlow.forEach(controlFlowNode -> controlFlowNode.addSuccessor(entryBlock.getSuccessor()));
-                current = Stream.concat(Stream.of(getControlFlowNodeMissingOneSuccessor(conditionNodes)), bodyAnalysis.breakFlow.stream()).collect(Collectors.toSet());
-            }
```

Comment:
I wrote all this logic yesterday. If we just handle `J.Empty` as if it's an AST node, then everything just works and all this extra stuff isn't required.